llama_index/schema.py

```diff
@@ -1,12 +1,14 @@
 """Base schema for data structures."""
+import uuid
 from abc import abstractmethod
-from dataclasses import dataclass
+from enum import Enum, auto
 from hashlib import sha256
-from typing import Any, Dict, List, Optional
+from pydantic import BaseModel, Field, root_validator
+from typing import Any, Dict, List, Optional, Union
 
-from dataclasses_json import DataClassJsonMixin
 
-from llama_index.utils import get_new_id
+DEFAULT_TEXT_NODE_TMPL = "{metadata_str}\n\n{content}"
+DEFAULT_METADATA_TMPL = "{key}: {value}"
 
 
 def _validate_is_flat_dict(metadata_dict: dict) -> None:
```

Comment:
is this still used?

Reply:
Yea, need to clean this up once I figure out which vector stores actually require this

Reply:
went through and added a spec for this to the vector stores that use metadata. Not every vector db documents the requirements for this, so I made an educated guess for some :)